### ISSUE TITLE
Server won't close when tab closes

### ISSUE THREAD

**Issue by darklight9811** [vercel/turbo#908]
### What version of Turborepo are you using?

1.1.3

### What package manager are you using / does the bug impact?

Yarn v1

### What operating system are you using?

Linux

### Describe the Bug

When I run dev (turbo run dev --parallel), if I close the IDE/terminal it still keeps the process running. Making the ports unavailable. Sometimes this happens even when I Ctrl + C the process.

### Expected Behavior

The process to finish when I close the terminal or kill it.

### To Reproduce

Call turbo run dev --parallel on a multiproject repo

**Comment by vasco3** [comment]
I'm having the same issue. @darklight9811 did you find a workaround?

**Comment by darklight9811** [comment]
@vasco3 on linux I find the specific process and close it, but usually it's a big hassle. So I just use another port (or don't declare it so will find the next open port automatically) or just restart the computer.

**Comment by vasco3** [comment]
ok, I'm using VSCode so when I restart VSCode it kills the hanging process

**Comment by darklight9811** [comment]
@vasco3 sadly that didn't work for me. Do you work on linux or windows? What vscode are you using? (if linux, from flatpak, from vscode official website), and it's version.

**Comment by vasco3** [comment]
@darklight9811 I'm in MacOS the latest with vscode from the official website.

I also notice that it doesn't closes the processes when I close the tab but it does the right thing when I kill the process with `Ctrl+C`

**Comment by makeryi** [comment]
> @darklight9811 I'm in MacOS the latest with vscode from the official website.
> 
> I also notice that it doesn't closes the processes when I close the tab but it does the right thing when I kill the process with `Ctrl+C`

me too

**Comment by glinda93** [comment]
Still reproducible with the following environment

Turborepo Version: 1.5.5
Package Manager: pnpm 
Operating System: Windows 11 + Git Bash
Child Tasks: Koa + Next.js dev in parallel

**Comment by willtemperley** [comment]
Also reproducible for me - Ctrl-C doesn't kill the servers, nor does stopping the terminal in VSCode. Restarting VSCode however does work.

Turborepo Version: 1.5.5
Package Manager: npm
Operating System: macOS Monterey
Child Tasks: vite & wrangler dev

**Comment by willtemperley** [comment]
Edit: Ctrl-C does kill all processes when using "turbo run dev --parallel"

**Comment by gsoltis** [comment]
Trying to sort out the symptoms here:
 - linux (or macos) commandline: `Ctrl+C` works as expected
 - running via vscode -> server processes stay alive if you exit vscode
 - running via GitBash + Windows 11 -> turbo exits, but server processes stay alive on `Ctrl+C`?

**Comment by willtemperley** [comment]
I'm not having any of those issues anymore. Tried with and without --parallel, in vscode and in terminal and all is working great (thanks for a great tool!). I'm on Mac.

**Comment by skreutzberger** [comment]
I am having the issue too under VSCode.

**Comment by Nazzanuk** [comment]
Yep this issue still exists, specifically this one 

> running `turbo dev` via npm sreipts in vscode -> server processes stay alive if you exit vscode (on Mac)

If you run it in the terminal the process seems to quit ok

**Comment by malkhuzayyim** [comment]
Running latest turbo version on Windows 10, inside VSCode with an integrated Gitbash terminal.

Repo has two apps, nuxtjs and nodejs api, running with `turbo run dev --parallel`

Quitting the process with Ctrl+C doesn't quit correctly, I have to kill the terminal and go delete the pid from turbo's temp directory to restart a clean process.

**Comment by j0hnm4r5** [comment]
Same issue here on macOS via VSCode's NPM Scripts panel.

I would expect all services to stop when restarting the script via that panel, but they don't, and I have to run `npx kill-port`.

**Comment by gsoltis** [comment]
Got it, thanks for the extra info. It sounds like this is a signal handling bug

**Comment by gsoltis** [comment]
@nathanhammond at least some of these cases (the windows ones) look similar to what you're looking into.

**Comment by williamlmao** [comment]
Any updates here?

**Comment by michischmidt** [comment]
I have the same problem on my M1 Mac in both VS Code and the Terminal

**Comment by vladutstoica** [comment]
Same issue here.
Turbo: v1.8.3
yarn: v1.22.19

Tested on Fleet, VSCode, and terminal.

**Comment by nathanhammond** [comment]
I'm closing this out as "completed" as of the 1.8.5 release and will be locking this thread. We very much appreciate your reports. We're closing it to help collate issues instead of having them tagged onto old issues.

If you find this issue in the future please file a new issue, *even if it produces symptoms identical to things described by this issue*! We're not trying to shut down conversation, just trying to make it easier to address all of the issues!
